### PR TITLE
Update scripts and yml file

### PR DESCRIPTION
**Pull request by rissangs**
1. fit_wine_quality_predict_model.py (@BruhatM )
>   - move main function up
>  - add plotting 
>  - add save_plots function

2. wine_ead.py (@wang-rui )
> - move main function up
>- add code in save_plots function to check output dir exists or not.

3. reports.rmd (@athy9193)
>  - change path for table1 and figure2 (haven't change any description of table and figure)

4. wine_env.yml
>  - add `catboost`

This PR refers to issue #40 , #41 , #43  and #47

### CODE REVIEW

**Review by athy9193 — COMMENTED** [review]
Hi @rissangs , can I check if we do not running StackingClassifier? Why do we need to install/import VotingClassifier, StackingClasssifer and CatBoost? If we do not, can we just omit this and do not change the yml?

**Comment by athy9193** [comment]
Regarding the report.md, I changing the links is sufficient, I will take care of the rest. No worries. Based on the assessment, we should have no conflict.

File: src/fit_wine_quality_predict_model.py [hunk 1]
```diff
@@ -21,12 +23,13 @@
...
 import numpy as np
 import pandas as pd
+import altair as alt
 
 # data
 from sklearn import datasets
 from sklearn.compose import ColumnTransformer, make_column_transformer
 from sklearn.dummy import DummyClassifier, DummyRegressor
-from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.ensemble import (RandomForestClassifier, RandomForestRegressor, VotingClassifier, StackingClassifier)
```
Comment (athy9193) [inline src/fit_wine_quality_predict_model.py]: Can we take the VotingClassifier, StackingClassifier out here?

File: src/fit_wine_quality_predict_model.py [hunk 2]
```diff
@@ -60,34 +63,14 @@
 from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
 from sklearn.model_selection import cross_val_predict
 from sklearn.metrics import plot_precision_recall_curve, plot_roc_curve
+from sklearn.svm import SVC
+from catboost import CatBoostClassifier
```
Comment (athy9193) [inline src/fit_wine_quality_predict_model.py]: Pls take CatBoost out as well. Thank you.

File: src/fit_wine_quality_predict_model.py [hunk 3]
```diff
@@ -111,16 +94,115 @@ def main(in_file_1, out_dir):
...
-    store_cross_val_results(f"preprocessing + MLP", scores_iter, results_df)
-    print(store_cross_val_results)
+    # Model selection
+    results={}
+    scoring_metric = {'f1_micro'}
+
+    pipe_rf = make_pipeline(preprocessor, RandomForestClassifier(bootstrap=False, max_depth=20,
+                                        max_features='sqrt', n_estimators=1800,
+                                        random_state=123))
+    pipe_catboost = make_pipeline(preprocessor, CatBoostClassifier(verbose=0, random_state=123))
```
Comment (athy9193) [inline src/fit_wine_quality_predict_model.py]: please remove pipe_catboost

File: src/fit_wine_quality_predict_model.py [hunk 4]
```diff
@@ -111,16 +94,115 @@ def main(in_file_1, out_dir):
...
+    pipe_svc = make_pipeline(preprocessor, SVC(probability=True))
+    classifiers = {
+        "svm": pipe_svc,
+        'random forest' : pipe_rf,
+        'CatBoost' : pipe_catboost,
+    }
+
+    stacking_model = StackingClassifier(list(classifiers.items()),
```
Comment (athy9193) [inline src/fit_wine_quality_predict_model.py]: Pls remove stacking_model

**Comment by rissangs** [comment]
@athy9193 

I have updated what you have mentioned.

**Pull request merged**